src/Wizard/Platform/Readline/ReadlinePlatform.ts

```diff
@@ -0,0 +1,147 @@
+      \rIf you are running the configuration as part of a POC, both of these should have been sent to you via your sales contact.
+    `);
+
+    await this.processTokens();
+
+    console.log(this.delimiter);
+
+    await this.processExternalCommunication();
+
+    console.log(this.delimiter);
+
+    await this.processInternalCommunication();
+
+    console.log(this.delimiter);
+    console.log('Communication diagnostics done.');
+  }
+
+  private async processTokens(): Promise<void> {
+    const repeaterId = await this.question('Please enter your Repeater ID');
+    const authToken = await this.question(
+      'Please enter your Repeater API Token'
+    );
+
+    console.log('\n');
+
+    await this.process('Verifying provided Token and Repeater ID', async () => {
+      try {
+        await this.tokens.writeTokens({ repeaterId, authToken });
+
+        return true;
+      } catch (err) {
+        return false;
+      }
+    });
+  }
+
+  private async processExternalCommunication(): Promise<void> {
+    console.log('Starting EXTERNAL communication diagnostics:\n');
+
+    await this.process(
+      'Validating that the connection to amq.nexploit.app at port 5672 is open',
+      async () => {
+        try {
+          return this.connectivityService.getConnectivityStatus(TestType.AUTH);
+        } catch (err) {
+          return false;
+        }
+      }
+    );
+    await this.process(
+      'Validating that the connection to nexploit.app at port 443 is open',
+      async () => {
+        try {
+          return this.connectivityService.getConnectivityStatus(TestType.HTTP);
+        } catch (err) {
+          return false;
+        }
+      }
+    );
+
+    console.log('\nEXTERNAL communication diagnostics completed.');
+  }
+
+  private async processInternalCommunication(): Promise<void> {
+    console.log(
+      'Next step is to validate the connection to your INTERNAL (local) target application(s).\n'
+    );
+
+    const urls = Helpers.getDelimitedInput(
+      await this.question(
+        'Please enter the target URLs to test (separated by commas)'
+      ),
+      ','
+    );
+
+    console.log(this.delimiter);
+
+    console.log('Starting INTERNAL communication diagnostics:\n');
+
+    for (const url of urls) {
+      await this.process(`Trying to reach ${url}`, async () => true);
+    }
+
+    console.log('\nEXTERNAL communication diagnostics completed.');
+    console.log('1 out of 3 URLs could not be reached.');
+  }
+
+  private async question(question: string): Promise<string> {
+    return new Promise((resolve) => {
+      this.rl.question(`${question}: `, resolve);
+    });
+  }
+
+  private async process(
+    text: string,
+    handler?: () => Promise<boolean>
+  ): Promise<void> {
+    process.stdout.write(`${text}...`);
+    readline.cursorTo(process.stdout, 0);
+
+    const result = await handler();
+
+    console.log(`${text}... ${result ? 'Success' : 'Failed'}`);
```

Comment:
I suggest to simplify callback and reduce duplicates by wrapping code into `try/catch` in `process` method

```suggestion
    process.stdout.write(`${text}...`);
    readline.cursorTo(process.stdout, 0);
    
    let result: boolean
    
    try {
        result = await handler();
    } catch {
        result =  false;
    }
    
    console.log(`${text}... ${result ? 'Success' : 'Failed'}`);
```

Reply:
It was my first implementation)) ok.